docs/site/src/components/apiMenuData.js

```diff
@@ -0,0 +1,68 @@
+// @flow weak
+
+// API menus for components whose entries don't follow `component`, `componentSub1`, `componentSub2` pattern.
+export const apiMenus = {
```

Comment:
Same as you, I don't understand why we can't keep a full mapping.
I think it's more about a design decision. Do we accept to have some magic in order to have less code?
I'm not convinced by the approach that tries to be clever with already three exceptions.